rio_tiler/colormap.py

```diff
@@ -133,6 +139,67 @@ def apply_discrete_cmap(
     return data[:-1], data[-1]
 
 
+def parse_color(value: Union[List[int], str]) -> List[int]:
+    """Parse RGB/RGBA color and return valid rio-tiler compatible RGBA colormap entry.
+
+    Args:
+        value (str or list of int): HEX encoded or list RGB or RGBA colors.
+
+    Returns:
+        list: RGBA values.
+
+    Examples:
+        >>> parse_color("#FFF")
+        [255, 255, 255, 255]
+
+        >>> parse_color("#FF0000FF")
+        [255, 0, 0, 255]
+
+        >>> parse_color("#FF0000")
+        [255, 0, 0, 255]
+
+        >>> parse_color([255, 255, 255])
+        [255, 255, 255, 255]
+
+    """
+    if isinstance(value, str):
+        if re.match("^#[a-fA-F0-9]{3,4}$", value):
```

Comment:
IMO a regex is overkill here. You can just assert that the string starts with `#` (if you want) and then if the string has <=4 more characters, you treat it as the first type, otherwise the second. And if an error is created that gets propagated to the user. But not a strong preference.

Reply:
FYI, that's how matplotlib does: https://github.com/matplotlib/matplotlib/blob/fa73bc09d62aef99842bbbd13dca687ac5bbd0b9/lib/matplotlib/colors.py#L237-L265

I think overkill regex are fine :-)